db.js

```diff
@@ -105,157 +108,307 @@ function addWatch(user, item, price, server) {
-        console.log(err);
-    })
 }
 
-function endWatch(user, item, server) {
-    // let item = unsanitizedItem.replace(/\'/g, "''");
-
-    findOrAddUser(user)
-    .then((results) => {
-        let userId = results;
-        findOrAddItem(item)
+function endWatch(user, item, server, watchId) {
+    if (watchId) {
+        const queryStr = `UPDATE watches SET active = false WHERE id = $1;`;
+        connection.query(queryStr, [watchId])
+            .catch(console.error);
+        
+    } else {
+        findOrAddUser(user)
         .then((results) => {
-            let itemId = results;
-            let queryStr = 'DELETE FROM watches WHERE user_id = $1 AND item_id = $2 AND server = $3';
-            connection.query(queryStr, [userId, itemId, server]);
-        })
-        .catch((err) => {
-            console.log(err);
+            let userId = results;
+            findOrAddItem(item)
+            .then((results) => {
+                let itemId = results;
+                let queryStr = `UPDATE watches SET active = false WHERE user_id = $1 AND item_id = $2 AND server = $3;`;
+                // FROM watches INNER JOIN users on watches.user_id = users.id INNER JOIN items on watches.item_id = items.id WHERE users.name = $1 AND FROM items WHERE items.name = $2 AND server = $3;';
+                connection.query(queryStr, [userId, itemId, server]).catch(console.error)
+            })
         })
-    })
-    .catch((err) => {
-        console.log(err);
-    })
+        .catch(console.error);
+    }
 }
 
 function endAllWatches(user) {
     findOrAddUser(user)
     .then((results) => {
         let userId = results;
-        let queryStr = 'DELETE FROM watches WHERE user_id = $1';
+        let queryStr = 'UPDATE watches SET active = false WHERE user_id = $1';
         connection.query(queryStr, [userId]);
     })
     .catch((err) => {
-        console.log(err);
+        console.error(err);
     })
 }
 
 function showWatch(user, item, callback) {
-    // console.log('db.showWatch item = ', item)
-    // let item = unsanitizedItem.replace(/\'/g, "''");
-
     findOrAddUser(user)
     .then((results) => {
         let userId = results;
-        findOrAddItem(item)
-        .then((results) => {
-            let itemId = results;
+        const pattern = '%'.concat(item).concat('%')
+        let queryStr = '' +
+            'SELECT watches.id, items.name, price, server, datetime, expiration, ' +
+            'CASE WHEN expiration IS NULL OR expiration < now() THEN FALSE ' +
+            'WHEN expiration >= now() THEN TRUE END AS snoozed ' +
+            'FROM watches ' +
+            'INNER JOIN items ON items.id = item_id ' +
+            'LEFT JOIN snooze_by_watch ON snooze_by_watch.watch_id = watches.id ' +
+            'WHERE watches.user_id = $1 AND items.name LIKE $2 AND watches.active = true'
+            'AND expiration IS NULL OR expiration > now() ORDER BY items.name ASC;';
+        connection.query(queryStr, [userId, pattern])
```

Comment:
I really hope this properly handles SQL Escaping, I will cry at you if this is a SQL Injection path, since all of the item names are user input.

Reply:
connection.query sanitizes the inputs in the second argument (the array) =)